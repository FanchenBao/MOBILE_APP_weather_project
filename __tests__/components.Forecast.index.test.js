import {Forecast} from '../src/components/Forecast/index.js';
import React from 'react';
import renderer from 'react-test-renderer';

it('renders correctly with city name and forecast', () => {
  const tree = renderer
    .create(
      <Forecast
        main="Clear"
        description="clear sky"
        temp="21.13"
        errorMsg=""
        name="Schenectady"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with error message', () => {
  const tree = renderer.create(<Forecast errorMsg="Some error" />).toJSON();
  expect(tree).toMatchSnapshot();
});
