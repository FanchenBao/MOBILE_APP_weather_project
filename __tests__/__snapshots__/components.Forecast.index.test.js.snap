// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with city name and forecast 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 40,
          },
          Object {
            "color": "#66ff33",
          },
        ]
      }
    >
      Schenectady
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "marginLeft": 15,
            },
            Object {
              "color": "#66ff33",
            },
          ]
        }
      >
        Condition:
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "fontWeight": "bold",
            },
            Object {
              "marginLeft": 15,
            },
            Object {
              "color": "#66ff33",
            },
          ]
        }
      >
        Temperature:
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-evenly",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "marginRight": 30,
            },
            Object {
              "color": "#66ff33",
            },
          ]
        }
      >
        clear sky
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 20,
            },
            Object {
              "marginRight": 30,
            },
            Object {
              "color": "#66ff33",
            },
          ]
        }
      >
        21.13 Â°F
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly with error message 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 40,
          },
          Object {
            "backgroundColor": "red",
            "borderRadius": 10,
            "color": "white",
            "fontSize": 20,
            "padding": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      Some error
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "flexDirection": "row",
      }
    }
  />
</View>
`;
